sdk/appcenter/src/test/java/com/microsoft/appcenter/persistence/DatabasePersistenceTest.java

```diff
@@ -388,22 +415,143 @@ public void checkSetStorageSizeForwarding() throws Exception {
 
         /* The real Android test for checking size is in DatabaseManagerAndroidTest. */
         DatabaseManager databaseManager = mock(DatabaseManager.class);
-        whenNew(DatabaseManager.class).withAnyArguments().thenReturn(databaseManager);
+        DatabasePersistence persistence = createDatabasePersistenceInstance(mock(Context.class), null, null, databaseManager, null);
         when(databaseManager.getCursor(any(SQLiteQueryBuilder.class), any(), any(String[].class), anyString()))
                 .thenReturn(mock(Cursor.class));
         when(databaseManager.setMaxSize(anyLong())).thenReturn(true).thenReturn(false);
 
         /* Just checks calls are forwarded to the low level database layer. */
-        DatabasePersistence persistence = new DatabasePersistence(mock(Context.class));
         assertTrue(persistence.setMaxStorageSize(20480));
         assertFalse(persistence.setMaxStorageSize(2));
     }
 
+    @Test(expected = PersistenceException.class)
+    public void failsToDeleteLogDuringPutWhenFull() throws Exception {
+        DatabaseManager databaseManager = mock(DatabaseManager.class);
+        DatabasePersistence persistence = createDatabasePersistenceInstance(mock(Context.class), null, null, databaseManager, null);
+
+        /* Set a mock log serializer. */
+        LogSerializer logSerializer = mock(LogSerializer.class);
+        String deserializedLog = "{name:\"MockLog\"}";
+        when(logSerializer.serializeLog(any(Log.class))).thenReturn(deserializedLog);
+        persistence.setLogSerializer(logSerializer);
+
+        /* Mock the database managers methods. */
+        when(databaseManager.getMaxSize()).thenReturn((long) deserializedLog.getBytes(StandardCharsets.UTF_8).length * 3);
+        when(databaseManager.deleteTheOldestRecord(anySet(), anyString(), anyInt())).thenReturn(null);
+        when(databaseManager.put(any(ContentValues.class))).thenThrow(new SQLiteFullException());
+
+        /* Persist a log. */
```

Comment:
Add comment about expected exception or comment/assert below this line that it should not execute successfully